### DISCUSSION TITLE
Issues with Rotating/Flipping GeomGrid in Pre-Processing tool

### DISCUSSION THREAD

**Discussion by 123fgh123** [damask-multiphysics/DAMASK#48] [Q&A] [answered]
Dear Colleagues,

I am experiencing difficulties with rotating/flipping a GeomGrid in the pre-processing stage of Damask. I have a 2D RVE file (with cell size 80x60x1) created by Dream3D from a real EBSD dataset, which is located in the X-Y plane by default and this plane may not be easily changed. To take advantage of parallelization during Damask simulations, I need to rotate the RVE to the X-Z plane, as parallelization only decomposes the simulation domain along the z-axis.

However, when I attempt to rotate or flip the RVE using the pre-processing tools (Damask 3.0.0 or 3.0.1), it does not seem to work. I have attached the pre-processing files for further examination. Despite setting the cell size to 80x1x60, the resulting cell size remains 80x60x1, which is identical to the input Dream3D file.
[pre-process.zip](https://github.com/user-attachments/files/18076155/pre-process.zip)

As an alternative, I would like to inquire about the possibility of parallelizing along the x-axis or y-axis. If this is feasible, I would no longer need to rotate the RVE. I believe parallelization is essential for my future simulations, which may involve larger cell sizes (e.g., 800x600x1).

Thank you for your assistance.

Best regards.

**Comment by eisenlohr** [comment, marked as answer]
Grid rotation appears to be working as expected:
```python
>>> import numpy as np
>>> import damask

>>> (g:=damask.GeomGrid(np.ones((80,60,1)),size=[80,60,1]))
cells:  80 × 60 × 1
size:   80.0 × 60.0 × 1.0 m³
origin: 0.0   0.0   0.0 m
# materials: 1 (min: 1, max: 1)

>>> (r:=g.rotate(damask.Rotation.from_axis_angle([1,0,0,90],degrees=True)))
cells:  80 × 1 × 60
size:   80.0 × 1.0 × 60.0 m³
origin: 0.0   29.5   -29.5 m
# materials: 1 (min: 1, max: 1)
```

Note that the `GeomGrid.rotate` method is **not in-place** but returns a new `GeomGrid`!

**Comment by 123fgh123** [comment]
I have a new try according to your suggestion, and it works well. 

Many thanks for your help!
